src/wallet/rpcwallet.cpp

```diff
@@ -2365,6 +2367,14 @@ UniValue listunspent(const UniValue& params, bool fHelp)
     if (params.size() > 1)
         nMaxDepth = params[1].get_int();
 
+    int nResultLimit = 0;
```

Comment:
Non-blocking: you could make this `boost::optional<size_t> nResultLimit = boost::none` directly, and then assign when parsing the parameter, so the `AvailableCoins` call is simpler.